### ISSUE TITLE
Pyright reports unbound variable when a TypedDict field shadows a builtin and the same builtin is used as a type annotation

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#2538]
**Describe the bug**

This bug is most easily described with a code example:

```py
class UserCountAggregateInput(TypedDict, total=False):
    id: bool
    bool: bool
    string: bool
```

2 errors are reported

```
error: "string" is unbound (reportUnboundVariable)
error: "bool" is unbound (reportUnboundVariable)
2 errors, 0 warnings, 0 infos
```

**To Reproduce**

Run pyright on strict mode

**Expected behavior**

No errors should be reported, the types at runtime are correctly evaluated.

```py
UserCountAggregateInput.__annotations__['string'] is bool
```

**VS Code extension or command-line**

Command-line: 1.1.184

**Comment by erictraut** [comment]
This behavior is correct. You have a namespace collision. You can work around it by doing the following:

```python
_bool = bool
class UserCountAggregateInput(TypedDict, total=False):
    id: _bool
    bool: _bool
    string: _bool
```

Or

```python
from builtins import bool as _bool

class UserCountAggregateInput(TypedDict, total=False):
    id: _bool
    bool: _bool
    string: _bool
```

Or

```python
import builtins


class UserCountAggregateInput(TypedDict, total=False):
    id: builtins.bool
    bool: builtins.bool
    string: builtins.bool
```
